src/Compilers/Core/Portable/CommandLine/CommonCompiler.ExistingReferencesResolver.cs

```diff
@@ -40,16 +38,16 @@ public ExistingReferencesResolver(MetadataReferenceResolver resolver, ImmutableA
                     from reference in _availableReferences
                     let identity = TryGetIdentity(reference)
                     where identity != null
-                    select identity));
+                    select identity!));
             }
 
-            public override ImmutableArray<PortableExecutableReference> ResolveReference(string reference, string baseFilePath, MetadataReferenceProperties properties)
+            public override ImmutableArray<PortableExecutableReference> ResolveReference(string reference, string? baseFilePath, MetadataReferenceProperties properties)
             {
                 var resolvedReferences = _resolver.ResolveReference(reference, baseFilePath, properties);
-                return resolvedReferences.WhereAsArray(r => _lazyAvailableReferences.Value.Contains(TryGetIdentity(r)));
+                return resolvedReferences.WhereAsArray(r => _lazyAvailableReferences.Value.Contains(TryGetIdentity(r)!));
```

Comment:
>TryGetIdentity(r)! [](start = 100, length = 18)

Will `TryGetIdentity(r)` always return non-`null`?

Reply:
No. It can absolutely return `null`. I didn't want to mix refactoring here with the annotations. I will file a follow up issue for figuring out how to write this better.

Reply:
https://github.com/dotnet/roslyn/issues/48954